pandas/core/reshape/merge.py

```diff
@@ -2356,7 +2356,7 @@ def _items_overlap_with_suffix(
     If corresponding suffix is empty, the entry is simply converted to string.
 
     """
-    if not is_list_like(suffixes, allow_sets=False):
+    if not (is_list_like(suffixes, allow_sets=False) or suffixes is None):
         warnings.warn(
             f"Passing 'suffixes' as a {type(suffixes)}, is not supported and may give "
             "unexpected results. Provide 'suffixes' as a tuple instead. In the "
```

Comment:
is this case now hit in tests?

Reply:
there are no tests that `warn` is called, not even if suffix isn't string like
should I add tests for that scenario?